code/modules/holodeck/computer.dm

```diff
@@ -110,207 +126,267 @@
-		derez(item, !force)
+/obj/machinery/computer/holodeck/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	emergency_shutdown()
 
-	program = A
-	// note nerfing does not yet work on guns, should
-	// should also remove/limit/filter reagents?
-	// this is an exercise left to others I'm afraid.  -Sayu
-	spawned = A.copy_contents_to(linked, 1, nerf_weapons = !(obj_flags & EMAGGED))
-	for(var/obj/machinery/M in spawned)
-		M.flags_1 |= NODECONSTRUCT_1
-	for(var/obj/structure/S in spawned)
-		S.flags_1 |= NODECONSTRUCT_1
-	effects = list()
+/obj/machinery/computer/holodeck/ex_act(severity, target)
+	emergency_shutdown()
+	return ..()
 
-	addtimer(CALLBACK(src, .proc/finish_spawn), 30)
+/obj/machinery/computer/holodeck/Destroy()
+	emergency_shutdown()
+	if(linked)
+		linked.linked = null
+		linked.power_usage = new /list(AREA_USAGE_LEN)
+	return ..()
 
-/obj/machinery/computer/holodeck/proc/finish_spawn()
-	var/list/added = list()
-	for(var/obj/effect/holodeck_effect/HE in spawned)
-		effects += HE
-		spawned -= HE
-		var/atom/x = HE.activate(src)
-		if(istype(x) || islist(x))
-			spawned += x // holocarp are not forever
-			added += x
-	for(var/obj/machinery/M in added)
-		M.flags_1 |= NODECONSTRUCT_1
-	for(var/obj/structure/S in added)
-		S.flags_1 |= NODECONSTRUCT_1
-	if(istype(program, /area/holodeck/rec_center/thunderdome1218) && !SSshuttle.shuttle_purchase_requirements_met[SHUTTLE_UNLOCK_MEDISIM])
-		say("Special note from \"1218 AD\" developer: I see you too are interested in the REAL dark ages of humanity! I've made this program also unlock some interesting shuttle designs on any communication console around. Have fun!")
-		SSshuttle.shuttle_purchase_requirements_met[SHUTTLE_UNLOCK_MEDISIM] = TRUE
+/obj/machinery/computer/holodeck/blob_act(obj/structure/blob/B)
+	emergency_shutdown()
+	return ..()
 
-/obj/machinery/computer/holodeck/proc/derez(obj/O, silent = TRUE, forced = FALSE)
-	// Emagging a machine creates an anomaly in the derez systems.
-	if(O && (obj_flags & EMAGGED) && !machine_stat && !forced)
-		if((ismob(O) || ismob(O.loc)) && prob(50))
-			addtimer(CALLBACK(src, .proc/derez, O, silent), 50) // may last a disturbingly long time
-			return
 
-	spawned -= O
-	if(!O)
-		return
-	var/turf/T = get_turf(O)
-	for(var/atom/movable/AM in O) // these should be derezed if they were generated
-		AM.forceMove(T)
-		if(ismob(AM))
-			silent = FALSE					// otherwise make sure they are dropped
+/obj/machinery/computer/holodeck/offstation //second holodeck if you want to add one to a ruin :flushed:
+	name = "holodeck control console"
+	desc = "A computer used to control a nearby holodeck."
+	offline_program = "holodeck_offline"
+	holodeck_access = CUSTOM_HOLODECK_ONE
+	mappedstartarea = /area/holodeck/rec_center/offstation_one
 
-	if(!silent)
-		visible_message("<span class='notice'>[O] fades away!</span>")
-	qdel(O)
+/obj/machinery/computer/holodeck/offstation/LateInitialize()
+	holodeck_access |= STATION_HOLODECK
+	. = ..()
```

Comment:
Any reason to call parent after instead of before?

Reply:
oops, misunderstood how to work with bitflags when i made that, theres no more lateinitialize in there now